### DISCUSSION TITLE
Can tiddlyclip store a fieldvalue from a tiddler while docking and use it in a tiddlyclip variable

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#131] [Q&A] [answered]
While docking tiddlers, can tiddlyclip acquire a fieldvalue from that tiddler ? Can this value be stored as a tiddlyclip variable and used in the any of the tiddlyclip rules ?

```
#rule=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#mediadir=Tiddlywiki filter (example {{!!sub-rad}}
$text=((*$text*))![](./((*@saveto(#astr,@clipsnap,#mediadir)*)))\n

```

**Comment by buggyj** [comment, marked as answer]
I think for the custom version of  'dock button' you are using the name of the docked tiddler is store in the tid $:/temp/tiddock

within your rule this tiddler can be accessed using the ':=' assignment, which is similar to transclusion

```
#dockedtid:={{$:/temp/tiddock}}
#field=somefield
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
```
....corrected

**Reply by arunnbabu81** [reply]
> I think for the custom version of 'dock button' you are using the name of the docked tiddler is store in the tid $:/temp/tiddock
> 
> within your rule this tiddler can be accessed using the ':=' assignment, which is similar to transclusion
> 
> ```
> #dockedtid:={{$:/temp/tiddock}}
> #field=somefield
> #fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
> ```
> 
> ....corrected

```
#rule=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#dockedtid:={{$:/temp/tiddock}}
#field=sub-rad
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
$text=((*$text*))![](./files/((*@saveto(#astr,@clipsnap,#fieldvalue)*)))\n
```

I did like this but its not working

**Reply by arunnbabu81** [reply]
Actually it is working. That field was not present in the tiddler in which i was testing

**Comment by arunnbabu81** [comment]
```
#field=sub-rad
```

Instead of this, can we use an equivalent of this tiddlywiki filter expression 
```
{{{ [{$:/HistoryList!!current-tiddler}fields[]prefix[sub-]] }}}
```

**Reply by arunnbabu81** [reply]
@buggyj  Is this possible?

**Reply by buggyj** [reply]
there is a filter macro:

```
#filterdir=[{$:/temp/tiddock}get[sub-rad]]
#dir=((*@runfilter(#filterdir)*))
```

or even

`#dir=((*@runfilter([{$:/temp/tiddock}get[sub-rad]])*))`

Personally I try to avoid filters as I have to 'run' the filters in my head to understand them

**Reply by arunnbabu81** [reply]
> there is a filter macro:
> 
> ```
> #filterdir=[{$:/temp/tiddock}get[sub-rad]]
> #dir=((*@runfilter(#filterdir)*))
> ```
> 
> or even
> 
> `#dir=((*@runfilter([{$:/temp/tiddock}get[sub-rad]])*))`
> 
> Personally I try to avoid filters as I have to 'run' the filters in my head to understand them

@buggyj why this one is not working

```
#rule=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#dockedtid:={{$:/temp/tiddock}}
#field=((*@runfilter([{$:/temp/tiddock}fields[]prefix[sub-]])*))
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
$text=((*$text*))![](./files/((*@saveto(#astr,@clipsnap,#fieldvalue)*)))\n

```

**Reply by buggyj** [reply]
may be  
`@runfilter([{$:/temp/tiddock}fields[]prefix[sub-]] `

returns a list

try removing  square brackets

```
#rule=/\[\[([\s\S]+)\]\]/
#term2=$1
#field=((*#field/#rule/#term2*))
```

**Reply by arunnbabu81** [reply]
> may be `@runfilter([{$:/temp/tiddock}fields[]prefix[sub-]] `
> 
> returns a list
> 
> try removing square brackets
> 
> ```
> #rule=/\[\[([\s\S]+)\]\]/
> #term2=$1
> #field=((*#field/#rule/#term2*))
> ```

Still not working

![image](https://github.com/user-attachments/assets/fa3e8a51-e90c-46ba-9a12-f562d7feb047)


Code used is given below
```
#rule1=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule1/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#dockedtid:={{$:/temp/tiddock}}
#field=((*@runfilter([{$:/temp/tiddock}fields[]prefix[sub-]])*))
#rule2=/\[\[([\s\S]+)\]\]/
#term2=$1
#field=((*#field/#rule2/#term2*))
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
$text=((*$text*))![](./files/((*@saveto(#astr,@clipsnap,#fieldvalue)*)))\n
```


and 

```
#rule=/\[\[([\s\S]+)\]\]/
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#dockedtid:={{$:/temp/tiddock}}
#field=((*@runfilter([{$:/temp/tiddock}fields[]prefix[sub-]])*))
#term2=$1
#field=((*#field/#rule/#term2*))
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
$text=((*$text*))![](./files/((*@saveto(#astr,@clipsnap,#fieldvalue)*)))\n

```

**Reply by buggyj** [reply]
```
#filter=[{$:/temp/tiddock}fields[]prefix[sub-]]
#field=((*@runfilter(#filter)*))
```

**Reply by arunnbabu81** [reply]
> ```
> #filter=[{$:/temp/tiddock}fields[]prefix[sub-]]
> #field=((*@runfilter(#filter)*))
> ```

This one works. The working code is given below

```
#rule=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
#dockedtid:={{$:/temp/tiddock}}
#filter=[{$:/temp/tiddock}fields[]prefix[sub-]]
#field=((*@runfilter(#filter)*))
#fieldvalue:={{((*#dockedtid*))!!((*#field*))}}
$text=((*$text*))![](./files/((*@saveto(#astr,@clipsnap,#fieldvalue)*)))\n

```

**Reply by arunnbabu81** [reply]
Are there any other hidden functions like this `runfilter `?
Is there any way to invoke action widgets ? Just asking

**Reply by buggyj** [reply]
There is a lot more to tiddlyclip!


actions are like this:

```
#action=<$action-setfield $tiddler="""((*@currentTid*))""" hidedesc="true"/>
#null=((*@doaction(#action)*))
```

**Reply by arunnbabu81** [reply]
> There is a lot more to tiddlyclip!
> 
> actions are like this:
> 
> ```
> #action=<$action-setfield $tiddler="""((*@currentTid*))""" hidedesc="true"/>
> #null=((*@doaction(#action)*))
> ```

Suppose I want to add different values to the field called `note-subject`  , only way I know of in tiddlyclip is to make use of the tags in the browser dropdown window as shown here
https://github.com/buggyj/tiddlyclip-plugin/discussions/59#discussion-3828289

![image](https://github.com/user-attachments/assets/d80ed604-fbd5-41d0-86d4-faea6b130468)

Is there any way to add the tag selected from the dropdown as fieldvalue to the field `note-subject` 

Can the dropdown add fields in a similar way as it adds the fields

Also I can't find the TiddlyclipOpts now.

**Reply by buggyj** [reply]
You can create the TiddlyClipOpts tiddler. In the options tiddler turn off the 'autoextratags' like so:

![image](https://github.com/user-attachments/assets/6ec4ec11-ede6-4a54-8a1d-b67a34ffc11a)


The selected tags get passed to the variable `@extraTags` - tags must not contain spaces. You can assign this to you field

`$note-subject=((*@extraTags*))`
